app/services/valid_courses_resolver.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class ValidCoursesResolver
```

Comment:
could we use the `CourseValidator` instead as I think it's doing something similar?